### PR TITLE
Add dashboard link to RootController

### LINKED ISSUE #807: Dashboard link is not present in root HATEOAS link page.
Just like there is a link for stream/definitions

```
_links: {
streams/definitions: {
href: "http://mlp-dataflow-server.cfapps.io/streams/definitions"
},
```
and so on, there should be a link to the dashboard.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Resolves #807

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RootController.java [hunk 1]
```diff
@@ -81,6 +81,7 @@ public RootController(EntityLinks entityLinks) {
 	@RequestMapping("/")
 	public ResourceSupport info() {
 		ResourceSupport resourceSupport = new ResourceSupport();
+		resourceSupport.add(new Link("/dashboard", "dashboard"));
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RootController.java]: I know this is not strictly part of this PR, but we should try to limit the occurence of the magic string "/dashboard".
So maybe introduce a CONSTANT for it in `UiController` and reference that here, and also in `OAuthSecurityConfiguration` and `BasicAuthSecurityConfiguration`

Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RootController.java]: Good idea! will fix that here.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/UiController.java [hunk 2]
```diff
@@ -28,9 +28,11 @@
 @Controller
 public class UiController {
 
-	@RequestMapping("/dashboard")
+	public static final String WEB_UI_INDEX_PAGE_NAME = "dashboard";
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/UiController.java]: I would make the starting slash part of the constant, that 1) makes more sense (it's a part, not just a "word") and 2) would simplify its usage quite a bit

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/security/OAuthSecurityConfiguration.java [hunk 3]
```diff
@@ -42,12 +43,12 @@ protected void configure(HttpSecurity http) throws Exception {
 		http.antMatcher("/**")
 		.authorizeRequests()
 			.antMatchers(
-				"/security/info**", "/login**", "/dashboard/logout-success-oauth.html",
-				"/dashboard/styles/**", "/dashboard/images/**", "/dashboard/fonts/**",
-				"/dashboard/lib/**").permitAll()
+					"/security/info**", "/login**", "/"+ UiController.WEB_UI_INDEX_PAGE_NAME +"/logout-success-oauth.html",
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/security/OAuthSecurityConfiguration.java]: creating a method that does the concatenantion (maybe even accepting varargs) would make this more readable IMO

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/security/OAuthSecurityConfiguration.java [hunk 4]
```diff
@@ -43,15 +45,23 @@ protected void configure(HttpSecurity http) throws Exception {
...
+					"/security/info**", "/login**", StringUtils.arrayToCommaDelimitedString(appendIndexPageRoute("/logout-success-oauth.html",
+							"/styles/**", "/images/**", "/fonts/**", "/lib/**"))).permitAll()
 			.anyRequest().authenticated()
 		.and().httpBasic()
-		.and().logout().logoutSuccessUrl("/"+ UiController.WEB_UI_INDEX_PAGE_NAME +"/logout-success-oauth.html")
+		.and().logout().logoutSuccessUrl(UiController.WEB_UI_INDEX_PAGE_ROUTE + "/logout-success-oauth.html")
 		.and().csrf().disable();
 	}
 
+	private String[] appendIndexPageRoute(String... patterns) {
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/security/OAuthSecurityConfiguration.java]: Will change as well, but this is not technically the "index" page route. It's the whole *path* to the UI app.

**Comment by ericbottard** [comment]
@ilayaperumalg WDYT about https://github.com/ericbottard/spring-cloud-dataflow/commit/45458378e08867db709386e813787551fe76ea90 ?

**Comment by ilayaperumalg** [comment]
@ericbottard looks much better.

**Comment by ericbottard** [comment]
> @ericbottard looks much better.
Thank you, I just turned 38.

**Comment by ericbottard** [comment]
All jokes aside, I will merge this then

**Comment by ericbottard** [comment]
Merged as 1d1ce57

**Comment by ericbottard** [comment]
Grrrr, I updated the actual link to dashboard just after I ran a full build, and actually introduced a test failure. Will fix in a new commit

**Pull request closed without merging**